### PR TITLE
Restore selection for the focused input

### PR DESCRIPTION
**Pull request by zeppelin**
@simonihmig Could you please review? :)

I've added a skipped test which would be rather straightforward to implement, but I thought rewriting a good portion of the service for such an edge case is an overkill.

### CODE REVIEW

File: tests/acceptance/form-rehydration-test.js [hunk 1]
```diff
@@ -43,4 +43,64 @@ test('it restores focus', async function(assert) {
...
+
+  this.application = startApp();
+
+  await visit('/');
+  await destroyApp(this.application);
+});
+
+
+// All input selections could be restored in theory, but it adds very little value in practice...
+skip(`it restores selection for all input elements`, async function(assert) {
```
Comment (simonihmig) [inline tests/acceptance/form-rehydration-test.js:82]: Tbh, I wasn't even aware that non-focused elements can have a selection, is that really the case? Do you think this should be supported (in the future)? If not, we should remove this test I think! Would stay around being skipped forever ;)

**Review by simonihmig — COMMENTED** [review]
Looks awesome! Thanks a lot! 

I would think we should remove the skipped test though before merging, unless we really want to support that case...

File: tests/acceptance/form-rehydration-test.js (hunk 1 above)
Comment (zeppelin) [reply tests/acceptance/form-rehydration-test.js:82]: > Tbh, I wasn't even aware that non-focused elements can have a selection, is that really the case?

Yes, for example when navigating around inputs using the tab key on the keyboard, all individual input's selection/caret position and is retained.

> Do you think this should be supported (in the future)?

Even if it's supported, one needs to have a reaaaally long running promise or slow loading JS assets to benefit, but even if that's the case, I think it's hardly noticable. What I was thinking while writing the test is that since browsers normally support this feature and the whole rehydration process should be as seamless as possible in theory, it's OK to mention this capability - yet, I don't want to invest time into making this a reality. Especially since this addon is a temporary solution until Ember/Glimmer most likely does this out-of-the-box. If someone wants to implement it, it's fine, the skipped test is a good place to start. But in the end this is your addon, so feel free to decide however you like :) I like to keep skipped tests as a reminder to myself for very low-priority features, but that's just me ;)

**Comment by simonihmig** [comment]
@zeppelin Merged and released as 0.1.2. Thanks again!

**Comment by zeppelin** [comment]
@simonihmig Thank you! I enjoyed a lot working on this project, I think it's fundamental to have rehydration in FastBoot apps to get proper UX.
